core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -436,12 +482,20 @@ object Chunk {
     Booleans(values, offset, length)
 
   final case class Booleans(values: Array[Boolean], offset: Int, length: Int)
-      extends Chunk[Boolean] {
+      extends Chunk[Boolean]
+      with KnownElementType[Boolean] {
     checkBounds(values, offset, length)
+    def elementClassTag = ClassTag.Boolean
     def size = length
     def apply(i: Int) = values(offset + i)
     def at(i: Int) = values(offset + i)
 
+    def copyToArray[O2 >: Boolean](xs: Array[O2], start: Int): Unit =
+      if (xs.isInstanceOf[Array[Boolean]])
+        System.arraycopy(values, offset, xs, start, length)
+      else
+        values.iterator.slice(offset, offset + length).copyToArray(xs, start)
```

Comment:
Actually you can just cast and use `arraycopy`. `Array[Boolean] <: Array[AnyRef]` in the JVM's typesystem.

Reply:
Hm:

```scala
scala> val xs = new Array[Any](10)
xs: Array[Any] = Array(null, null, null, null, null, null, null, null, null, null)

scala> System.arraycopy(xs.asInstanceOf[Array[Byte]], 0, Array[Byte](1, 2, 3), 0, 3)
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [B
  ... 29 elided
```

Reply:
I… do not have a brain. :-D Not sure what I was thinking on this one.